azurerm/internal/services/desktopvirtualization/virtual_desktop_application_group_resource_test.go

```diff
@@ -0,0 +1,247 @@
+	}
+}
+
+func testCheckAzureRMDesktopVirtualizationApplicationGroupDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).DesktopVirtualization.ApplicationGroupsClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_virtual_desktop_application_group" {
+			continue
+		}
+
+		log.Printf("[WARN] azurerm_virtual_desktop_application_group still exists in state file.")
+
+		id, err := parse.VirtualDesktopApplicationGroupID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		result, err := client.Get(ctx, id.ResourceGroup, id.Name)
+		if err == nil {
+			return fmt.Errorf("Virtual Desktop Host Pool still exists:\n%#v", result)
+		}
+
+		if result.StatusCode != http.StatusNotFound {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureRMVirtualDesktopApplicationGroup_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-vdesktop-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_desktop_host_pool" "test" {
+  name                = "acctestHP"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  type                = "Pooled"
+  load_balancer_type  = "BreadthFirst"
+}
+
+resource "azurerm_virtual_desktop_application_group" "test" {
+  name                = "acctestAG%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  type                = "Desktop"
+  host_pool_id        = azurerm_virtual_desktop_host_pool.test.id
+}
+
+`, data.RandomInteger, data.Locations.Secondary, data.RandomIntOfLength(8))
+}
+
+func testAccAzureRMVirtualDesktopApplicationGroup_complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-vdesktop-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_desktop_host_pool" "test" {
+  name                 = "acctestHP"
+  location             = azurerm_resource_group.test.location
+  resource_group_name  = azurerm_resource_group.test.name
+  validate_environment = true
+  description          = "Acceptance Test: A host pool"
+  type                 = "Pooled"
+  load_balancer_type   = "BreadthFirst"
+}
+
+resource "azurerm_virtual_desktop_application_group" "test" {
+  name                = "acctestAG%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  type                = "Desktop"
+  host_pool_id        = azurerm_virtual_desktop_host_pool.test.id
+  friendly_name       = "TestAppGroup"
+  description         = "Acceptance Test: An application group"
+  tags = {
+    purpose = "acceptance-testing"
```

Comment:
could we ensure capitals work here
```suggestion
    Purpose = "Acceptance-Testing"
```